index.js

```diff
@@ -211,21 +211,21 @@ DecodingTransformer.prototype._handleCircularSelfRefDuringTransform = function (
     // new circular references. As a workaround we create getter, so once transformation
     // complete, dereferenced property will point to correct transformed object.
     var references = this.references;
+    var val = void 0;
 
     Object.defineProperty(parent, key, {
-        val:          void 0,
         configurable: true,
         enumerable:   true,
 
         get: function () {
-            if (this.val === void 0)
-                this.val = references[refIdx];
+            if (val === void 0)
+                val = references[refIdx];
 
-            return this.val;
+            return val;
         },
 
         set: function (value) {
-            this.val = value;
+            val = value;
```

Comment:
lgtm, thanks. But I've just realised that setter was implemented incorrectly all this time - assignment expression should return assigned value:
```js
val = value;
return val;
```
Could you add this fix as well please?

Reply:
Added to the PR